### PR TITLE
fix farm name validation special chars and min length

### PR DESCRIPTION
**Pull request by amiraabouhadid**
### Changes

add allow only -_ as special characters, make min length 3 chars to be consistent with tfchain validation

### Related Issues

#1686

### CODE REVIEW

File: packages/playground/src/dashboard/components/create_farm.vue [hunk 1]
```diff
@@ -92,11 +94,17 @@ export default {
         isCreating.value = false;
       }
     }
+    function validateFarmName(name: string) {
+      if (!name.split("").every((c: string) => /[a-zA-Z0-9\-_]/.test(c))) {
+        return { message: "Invalid Farm Name" };
```
Comment (0oM4R) [inline packages/playground/src/dashboard/components/create_farm.vue:99]: I suggest to make the message contain the name rules,  what characters are allowed and so on

Comment (amiraabouhadid) [reply packages/playground/src/dashboard/components/create_farm.vue]: done, thanks

File: packages/playground/src/dashboard/components/create_farm.vue [hunk 2]
```diff
@@ -19,11 +19,13 @@
...
                   validators.required('Farm name is required.'),
                   name => validators.isAlpha('Farm name must start with an alphabet char.')(name[0]),
-                  validators.minLength('Farm name minimum length is 2 chars.', 2),
+                  validators.minLength('Farm name minimum length is 3 chars.', 3),
                   validators.maxLength('Farm name maximum length is 40 chars.', 40),
                   validators.pattern('Farm name  should not contain whitespaces.', {
                     pattern: /^[^\s]+$/,
                   }),
+
+                  validateFarmName,
```
Comment (MohamedElmdary) [inline packages/playground/src/dashboard/components/create_farm.vue:28]: can we use validators.pattern like the above one ?

Comment (0oM4R) [reply packages/playground/src/dashboard/components/create_farm.vue:28]: and please remove the extra space in line 24 after `Farm name`

Comment (amiraabouhadid) [reply packages/playground/src/dashboard/components/create_farm.vue:28]: validators.pattern checks only if pattern exists, it does not check if every character satisfies pattern.

File: packages/playground/src/dashboard/components/create_farm.vue [hunk 3]
```diff
@@ -92,11 +94,17 @@ export default {
         isCreating.value = false;
       }
     }
+    function validateFarmName(name: string) {
+      if (!name.split("").every((c: string) => /[a-zA-Z0-9\-_]/.test(c))) {
+        return { message: "Farm name can only contain alphabetic letters, numbers, '-' or '_'" };
+      }
+    }
```
Comment (zaelgohary) [inline packages/playground/src/dashboard/components/create_farm.vue:101]: Can we use the pattern method from validators instead?

**Review by maayarosama — APPROVED** [review]

**Review by AlaaElattar — APPROVED** [review]

**Review by 0oM4R — APPROVED** [review]

**Pull request merged**